functions/src/aqi-calculation/util.ts

```diff
@@ -1,69 +1,56 @@
-
-  const purpleAirApiResponse = await axios({
-    url: PURPLE_AIR_API_ADDRESS,
-    params: {
-      show: purpleAirId,
-    },
-  });
-
-  return new PurpleAirResponse(purpleAirApiResponse);
-}
+const readingsSubcollection: (docId: string) => string = (docId: string) =>
+  `/sensors/${docId}/readings`;
 
 /**
- * For a given readings subcollection, gets the most recent reading Timestamp
- * @param readingsCollectionRef - reference to readings collection for sensor to get the most recent reading time
- * @returns a Promise of the Timestamp of the most recent sensor reading time, or null if no readings
+ * Full reading information for a PurpleAir sensor from the PurpleAir API
+ * - `name` - name of the sensor
+ * - `id` - PurpleAir ID of the sensor, also called the sensor_index. This is
+ *   the main identifier for a sensor.
+ * - `latitude` - latitude of a sensor
+ * - `longitude` - longitude of a sensor
+ * - `pm25` - PM 2.5 reading for a sensor. This value is the average of the
+ *   PM 2.5 reading for channelA and channelB
+ * - `humidity` - humidity reading for a sensor
+ * - `meanPercentDifference` - mean percent difference between the pseudo averages
+ *   of the readings for channelA and channelB, as calculated from the confidence
+ *   value returned by the PurpleAir API
+ * - `timestamp` - the timestamp of the current reading
  */
-async function getLastSensorReadingTime(
-  readingsCollectionRef: FirebaseFirestore.CollectionReference<FirebaseFirestore.DocumentData>
-): Promise<FirebaseFirestore.Timestamp | null> {
-  let lastSensorReadingTime: FirebaseFirestore.Timestamp | null = null;
-  const maxDocs = 1;
-  const querySnapshot = await readingsCollectionRef
-    .orderBy('timestamp', 'desc')
-    .limit(maxDocs)
-    .get();
-  // There should only be one document in docs, but loops over docs since it's an array
-  for (const sensorDoc of querySnapshot.docs) {
-    if (sensorDoc.data().timestamp) {
-      lastSensorReadingTime = sensorDoc.data().timestamp;
-    }
-  }
-  return lastSensorReadingTime;
+interface PurpleAirReading {
+  name: string;
+  id: number;
+  latitude: number;
+  longitude: number;
+  pm25: number;
+  humidity: number;
+  meanPercentDifference: number;
+  timestamp: Date;
 }
 
 /**
  * Sensor reading data that is stored in the readings subcollection
  * - `latitude` - latitude of a sensor
  * - `longitude` - longitude of a sensor
- * - `channelAPm25` - PM2.5 reading for channel A
- * - `channelBPm25` - PM2.5 reading for channel B
+ * - `pm25` - PM 2.5 reading for a sensor. This value is the average of the
+ *   PM 2.5 reading for channelA and channelB
  * - `humidity` - humidity reading for a sensor
+ * - `meanPercentDifference` - mean percent difference between the pseudo averages
+ *   of the readings for channelA and channelB, as calculated from the confidence
+ *   value returned by the PurpleAir API
  * - `timestamp` - the timestamp of the current reading
  */
 interface HistoricalSensorReading {
```

Comment:
The word historical feels overloaded now: does it refer to the old format or the per reading data (i know it means the per reading data and the comment says that, but I'm not sure if there's a better name)

Reply:
How about something like `StoredSensorReading`? `FirestoreSensorReading`?

Reply:
I'm leaving as is, but when we document the website, we need to add that we use the term "historical" to describe the readings subcollection historical database for each sensor.